prost-build/src/code_generator.rs

```diff
@@ -1068,6 +1052,47 @@ impl<'a> CodeGenerator<'a> {
         }
     }
 
+    /// Returns whether the Rust type for this field needs to be `Box<_>`.
+    ///
+    /// This can be explicitly configured with `Config::boxed`, or necessary
+    /// to prevent an infinitely sized type definition in case when the type of
+    /// a non-repeated message field transitively contains the message itself.
+    fn boxed(
+        &self,
+        field: &FieldDescriptorProto,
+        fq_message_name: &str,
+        oneof: Option<&str>,
+    ) -> bool {
+        let repeated = field.label == Some(Label::Repeated as i32);
+        let fd_type = field.r#type();
+        if !repeated
+            && (fd_type == Type::Message || fd_type == Type::Group)
+            && self
+                .message_graph
+                .is_nested(field.type_name(), fq_message_name)
+        {
+            return true;
+        }
+        let config_path = match oneof {
+            None => Cow::Borrowed(fq_message_name),
+            Some(ooname) => Cow::Owned(format!("{fq_message_name}.{ooname}")),
+        };
+        if self
+            .config
+            .boxed
+            .get_first_field(&config_path, field.name())
+            .is_some()
+        {
+            println!(
+                "cargo:warning=\
+                Field X is repeated and manually marked as boxed. \
+                This is deprecated and support will be removed in a later release"
+            );
```

Comment:
This is incorrect. The warning will show for any field marked as boxed.

```suggestion
            if repeated {
                println!(
                    "cargo:warning=\
                    Field X is repeated and manually marked as boxed. \
                    This is deprecated and support will be removed in a later release"
                );
            }
```